src/reconcile/rec_write.c

```diff
@@ -50,6 +50,11 @@ typedef struct {
 	bool leave_dirty;
 
 	/*
+	 * Track if an update was copied onto a page during reconciliation.
+	 */
+	bool update_used;
+
+	/*
```

Comment:
I originally called it `progress`.  It's tracking whether a `WT_UPDATE` is used when writing a new page image.  I honestly can't see why any eviction should go ahead if we don't set this, but constrain the check to rewrites for now because constant rewriting of pages is the worst case we see in real workloads (generally when pages grow larger than `memory_page_max` for various reasons).

I'll rename it to something clearer.